### ISSUE TITLE
[feature request] ignore_index and size_average in nn.AdaptiveLogSoftMaxWithLoss

### ISSUE THREAD

**Issue by aconneau** [pytorch/pytorch#12675]
**Feature request for nn.AdaptiveLogSoftMaxWithLoss**

Linked to https://github.com/pytorch/pytorch/pull/5287:

It would be nice if we could add these parameters to nn.AdaptiveLogSoftMaxWithLoss:

1) the "ignore_index" parameter, like in [F.cross_entropy](https://pytorch.org/docs/stable/nn.html?highlight=cross%20entropy#torch.nn.functional.cross_entropy), for batches with padding indexes
2) the size_average parameter

Thanks!
cc @SsnL @glample

**Comment by SsnL** [comment]
Btw, I changed (2) to the modern `reduction` parameter :)

**Comment by rohan-varma** [comment]
@SsnL @aconneau I'd love to tackle this issue if nobody is working on it. It looks like the relevant files to look at would be the current implementation `torch/nn/modules/adaptive.py` and how `ignore_index` works in `torch/nn/functional.py`?
